ghost/members-api/lib/repositories/member.js

```diff
@@ -1232,21 +1273,37 @@ module.exports = class MemberRepository {
                     from_plan: subscriptionModel.get('plan_id')
                 }, sharedOptions);
 
-                if (this._labsService.isSet('emailAlerts')) {
-                    const subscriptionPriceData = _.get(updatedSubscription, 'items.data[0].price');
-                    let ghostProduct;
-                    try {
-                        ghostProduct = await this._productRepository.get({stripe_product_id: subscriptionPriceData.product}, {...sharedOptions, forUpdate: true});
-                    } catch (e) {
-                        ghostProduct = null;
-                    }
-                    await this.staffService.notifyPaidSubscriptionCancel({
-                        member: member.toJSON(),
-                        subscription: updatedSubscription,
-                        cancellationReason: data.subscription.cancellationReason,
-                        tier: ghostProduct?.toJSON()
-                    });
+                const subscriptionPriceData = _.get(updatedSubscription, 'items.data[0].price');
+                let ghostProduct;
+                try {
+                    ghostProduct = await this._productRepository.get({stripe_product_id: subscriptionPriceData.product}, {...sharedOptions, forUpdate: true});
+                } catch (e) {
+                    ghostProduct = null;
                 }
+
+                const subscriptionData = {
+                    id: data.subscription.subscription_id,
+                    amount: subscriptionPriceData?.unit_amount,
+                    currency: subscriptionPriceData?.currency,
+                    interval: subscriptionPriceData?.recurring?.interval,
+                    cancelAt: updatedSubscription.cancel_at,
+                    canceledAt: updatedSubscription.canceled_at
+                };
+                const tierData = ghostProduct ? {
+                    id: ghostProduct.get('id'),
+                    name: ghostProduct.get('name')
+                } : null;
+
+                const context = options?.context || {};
+                const source = this._resolveContextSource(context);
+
+                DomainEvents.dispatch(MemberSubscriptionCancellationEvent.create({
```

Comment:
As above - do you mind if we just pass ids for now? I'm concerned about having different data shapes used throughout the app, this event pattern is still fairly new and hasn't quite settled yet.

Reply:
👍